src/test/java/dev/zarr/zarrjava/ZarrTest.java

```diff
@@ -1,217 +1,473 @@
+
+        BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
+        String line;
+        while ((line = reader.readLine()) != null) {
+            System.out.println(line);
+        }
+
+        BufferedReader readerErr = new BufferedReader(new InputStreamReader(process.getErrorStream()));
+        while ((line = readerErr.readLine()) != null) {
+            System.err.println(line);
+        }
+
+        int exitCode = process.waitFor();
+        assert exitCode == 0;
+
+        Array array = Array.open(new FilesystemStore(TESTOUTPUT).resolve("read_from_zarrita", codec));
+        ucar.ma2.Array result = array.read();
+
+        //for expected values see zarrita_write.py
+        Assertions.assertArrayEquals(new int[]{16, 16}, result.getShape());
+        Assertions.assertEquals(DataType.INT32, array.metadata.dataType);
+        Assertions.assertArrayEquals(new int[]{2, 8}, array.metadata.chunkShape());
+        Assertions.assertEquals(42, array.metadata.attributes.get("answer"));
+
+        int[] expectedData = new int[16 * 16];
+        Arrays.setAll(expectedData, p -> p);
+        Assertions.assertArrayEquals(expectedData, (int[]) result.get1DJavaArray(ucar.ma2.DataType.INT));
+    }
+
+    @CsvSource({"0,true", "0,false", "5, true", "10, false"})
+    @ParameterizedTest
+    public void testZstdLibrary(int clevel, boolean checksumFlag) throws IOException, InterruptedException {
+        //compress using ZstdCompressCtx
+        int number = 123456;
+        byte[] src = ByteBuffer.allocate(4).putInt(number).array();
+        byte[] compressed;
+        try (ZstdCompressCtx ctx = new ZstdCompressCtx()) {
+            ctx.setLevel(clevel);
+            ctx.setChecksum(checksumFlag);
+            compressed = ctx.compress(src);
+        }
+        //decompress with Zstd.decompress
+        long originalSize = Zstd.decompressedSize(compressed);
+        byte[] decompressed = Zstd.decompress(compressed, (int) originalSize);
+        Assertions.assertEquals(number, ByteBuffer.wrap(decompressed).getInt());
+
+        //write compressed to file
+        String compressedDataPath = TESTOUTPUT.resolve("compressed" + clevel + checksumFlag + ".bin").toString();
+        try (FileOutputStream fos = new FileOutputStream(compressedDataPath)) {
+            fos.write(compressed);
+        }
+
+        //decompress in python
+        Process process = new ProcessBuilder(
+                pythonPath(),
+                PYTHON_TEST_PATH.resolve("zstd_decompress.py").toString(),
+                compressedDataPath,
+                Integer.toString(number)
+        ).start();
+        int exitCode = process.waitFor();
+        assert exitCode == 0;
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"blosc", "gzip", "zstd", "bytes", "transpose", "sharding_start", "sharding_end", "crc32c"})
+    public void testWriteToZarrita(String codec) throws IOException, ZarrException, InterruptedException {
```

Comment:
We could probably merge this with testCodecsWriteRead to avoid some code duplication.